.github/workflows/Release.yml

```diff
@@ -0,0 +1,57 @@
+name: Release CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - "Project.toml"
```

Comment:
When this workflow is changed on a PR you should automatically trigger as much of this workflow as possible. Only triggering this workflow on `main` means any flaws in this logic aren't caught until after the changes are merged. For this workflow we just have to avoid the `push` operation which I would make into a separate step and add a conditional `if: github.event_name != 'pull_request'`

Reply:
good point - I've refactored the job pushing all "state changing" behaviour to the last step that is guarded by the `event_name` check